src/pages/cli/migration/lazy-load-custom-selection-set.mdx

```diff
@@ -0,0 +1,361 @@
+* **Models** now have extended support for lazy loading connected models to include has-one and belongs-to relations.
+* **DataStore (Swift)** now has better support for cross platform app development. Model updates from Amplify Studio and the Amplify Libraries including JavaScript and Android can be now observed in real-time by DataStore Swift apps.
+* **DataStore (Swift)** now supports the bi-directional has-one data modeling use case.
+
+
+## Custom Selection Sets
+
+Developers using **API (GraphQL)** can control the data returned from their GraphQL service. The request for a particular model can include or exclude connected models. Let’s take a look at the schema with Post and Comment models in the following examples. A comment belongs to a post and a post has many comments.
+
+```
+enum PostStatus {
+  ACTIVE
+  INACTIVE
+}
+
+type Post @model @auth(rules: [{allow: public}]) {
+  id: ID!
+  title: String!
+  rating: Int!
+  status: PostStatus!
+  comments: [Comment] @hasMany
+}
+
+type Comment @model @auth(rules: [{allow: public}]) {
+  id: ID!
+  content: String
+  post: Post @belongsTo
+}
+```
+
+Currently, developers querying for the Comment will contain the Post eager loaded:
+
+<BlockSwitcher>
+<Block name="Swift">
+
+```swift
+let response = try await Amplify.API.query(request: .get(Comment.self, byId: "commentId"))
+if case .success(let queriedComment) = response {
+    print("Queried Comment eager loaded post: \(queriedComment?.post)")
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+With the new model types and library changes, the same request will no longer eager load the post. The post is lazy loaded from the GraphQL service at the time the post is accessed.
+
+<BlockSwitcher>
+<Block name="Swift">
+
+```swift
+let response = try await Amplify.API.query(request: .get(Comment.self, byId: "commentId"))
+if case .success(let queriedComment) = response {
+    print("Queried Comment \(String(describing: queriedComment))")
+    // Lazy Load the post
+    if let post = try await queriedComment?.post {
+        print("Lazy loaded the post: \(post)")
+    }
+}
+```
+
+</Block>
+</BlockSwitcher>
+
+To achieve the previous behavior, specifying the model path using the new `includes` parameter:
+
+<BlockSwitcher>
+<Block name="Swift">
+
+```swift
+let response = try await Amplify.API.query(request:
+        .get(Comment.self,
+             byId: "commentId",
+             includes: { comment in [comment.post]} ))
+if case .success(let queriedComment) = response {
+    print("Queried Comment \(String(describing: queriedComment))")
+
+    if let post = try await queriedComment?.post {
+        print("Eager loaded post: \(post)")
+    }
+}
+```
+
+This will populate the Selection Set of the post in the GraphQL document which indicates to your GraphQL service to retrieve the post model as part of the operation. Once you await on the post, the post model will immediately be returned without making a service call.
```

Comment:
```suggestion
This will populate the selection set of the post in the GraphQL document which indicates to your GraphQL service to retrieve the post model as part of the operation. Once you await on the post, the post model will immediately be returned without making a service call.
```